File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
answer = [0]
def solution(string: str):
    cache = ''
    if len(string) == 1:
        answer[0] += 1
        return
    for idx in range(1, len(string)):
        if string[idx] in cache:
            return
        if string[idx - 1] != string[idx]:
            cache += string[idx - 1]
    answer[0] += 1

loop = int(input())
for _ in range(loop):
    solution(input())
print(answer[0])

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;


public class Main {

    public static int solution(int N, String[] words) {
        int answer = 0;
        for (String word : words) {
            int[] cache = new int[26];
            cache[word.charAt(0) - 97] = 1;
            boolean breaked = false;
            for (int i = 1; i < word.length(); i++) {
                if (word.charAt(i) != word.charAt(i - 1)) {
                    if (cache[word.charAt(i) - 97] == 1) {
                        breaked = true;
                        break;
                    }
                    cache[word.charAt(i) - 97] = 1;
                }
            }
            if (!breaked) {
                answer += 1;
            }
        }
        return answer;
    }

    public static void main(String[] args) throws IOException {
        InputStream input = System.in;
        BufferedReader br = new BufferedReader(new InputStreamReader(input));

        int N = Integer.parseInt(br.readLine());
        String[] words = new String[N];
        for (int i = 0; i < N; i++) {
            words[i] = br.readLine();
        }
        System.out.println(solution(N, words));
    }
}